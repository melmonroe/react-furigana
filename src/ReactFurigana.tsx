import Word from './components/Word.tsx';

const DEFAULT_SEPARATOR = ' ';
const DEFAULT_LEFT_BRACKET = '[';
const DEFAULT_RIGHT_BRACKET = ']';

type ReactFuriganaProps = {
    text: string,
    separator?: string,
    leftBracket?: string,
    rightBracket?: string
}

export function ReactFurigana(props: ReactFuriganaProps) {

    const separator = props.separator ?? DEFAULT_SEPARATOR;
    const leftBracket = props.leftBracket ?? DEFAULT_LEFT_BRACKET;
    const rightBracket = props.rightBracket ?? DEFAULT_RIGHT_BRACKET;

    return (
        <>
            {props.text.split(separator).map((word, index) => (
                <Word key={index} word={word} leftBracket={leftBracket} rightBracket={rightBracket}/>
            ))}
        </>
    );

}
