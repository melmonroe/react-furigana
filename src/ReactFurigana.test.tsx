import {render} from '@testing-library/react';
import {describe, expect, it} from 'vitest';
import ReactFurigana from './index.ts';

describe('ReactFurigana', () => {

    it('renders', () => {
        expect(render(
            <ReactFurigana text="日本 語[ご]の漢字[かんじ]は難[むずか]しい。"/>
        )).toMatchSnapshot();
    });

    it('々 renders', () => {
        expect(render(
            <ReactFurigana text="様々[さまざま]"/>
        )).toMatchSnapshot();
    });

    it('renders with a custom render function', () => {
        expect(render(
            <ReactFurigana
                text="日本語[にほんご]"
                render={token => (
                    <div>
                        <span className="kanji">
                            {token.kanji}
                        </span>
                            <span className="furigana">
                            {token.furigana}
                        </span>
                    </div>
                )}
            />
        )).toMatchSnapshot();
    });

});
