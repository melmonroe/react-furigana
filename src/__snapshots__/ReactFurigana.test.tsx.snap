// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ReactFurigana > renders 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      日本
      <ruby>
        語
        <rt>
          ご
        </rt>
      </ruby>
      の
      <ruby>
        漢字
        <rt>
          かんじ
        </rt>
      </ruby>
      は
      <ruby>
        難
        <rt>
          むずか
        </rt>
      </ruby>
      し
      い
      。
    </div>
  </body>,
  "container": <div>
    日本
    <ruby>
      語
      <rt>
        ご
      </rt>
    </ruby>
    の
    <ruby>
      漢字
      <rt>
        かんじ
      </rt>
    </ruby>
    は
    <ruby>
      難
      <rt>
        むずか
      </rt>
    </ruby>
    し
    い
    。
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ReactFurigana > renders with a custom render function 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <span
          class="kanji"
        >
          日本語
        </span>
        <span
          class="furigana"
        >
          にほんご
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <span
        class="kanji"
      >
        日本語
      </span>
      <span
        class="furigana"
      >
        にほんご
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ReactFurigana > 々 renders 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ruby>
        様々
        <rt>
          さまざま
        </rt>
      </ruby>
    </div>
  </body>,
  "container": <div>
    <ruby>
      様々
      <rt>
        さまざま
      </rt>
    </ruby>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
